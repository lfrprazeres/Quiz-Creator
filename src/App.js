import React from 'react';
import Routes from './routes';
import styled from 'styled-components';
import waveSvg from './assets/wave.svg';

// creating the wave in the background
const AppStyled = styled.main`
  background-image: url(${waveSvg});
  background-size: contain;
  background-position: center bottom;
  background-repeat: no-repeat;
  background-attachment: fixed;
  min-height: 100vh;
`

function App() {
  return (
    <AppStyled>
      <Routes />
    </AppStyled>
  );
}

export default App;
