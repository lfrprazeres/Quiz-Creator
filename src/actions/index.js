export {
    createQuiz,
    CREATE_QUIZ,

    createQuestions,
    CREATE_QUESTIONS,

    DELETE_QUIZ,
    onDeleteQuiz
} from './quiz'