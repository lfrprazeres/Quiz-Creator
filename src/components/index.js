import QuizTable from './QuizTable';
import CreateSection from './CreateSection';
import Button from './Button';

export {
    QuizTable,
    CreateSection,
    Button,
}