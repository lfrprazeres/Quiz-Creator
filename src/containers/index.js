import QuizList from './QuizList';
import NewQuiz from './NewQuiz';
import NewQuestions from './NewQuestions'
import AnswerQuestions from './AnswerQuestions';

export {
    QuizList,
    NewQuiz,
    NewQuestions,
    AnswerQuestions
}