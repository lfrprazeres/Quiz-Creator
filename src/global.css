@import url('https://fonts.googleapis.com/css?family=Lato:400,900&display=swap');

* {
    margin: 0;
    padding: 0;
    box-sizing: border-box;
    font-family: 'Lato', sans-serif;
}

body {
  background-color: #fafafa;
}

a {
  color: black;
  text-decoration: none;
}