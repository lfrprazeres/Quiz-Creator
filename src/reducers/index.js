import { combineReducers } from 'redux';
import quizReducer from './quizReducer';
import appReducer from './appReducer';

export const rootReducer = combineReducers({
    quiz: quizReducer,
    app: appReducer
})