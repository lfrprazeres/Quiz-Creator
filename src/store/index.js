import { createStore } from 'redux'
import { rootReducer } from '../reducers';
import { persistStore, persistReducer } from 'redux-persist'
import storage from 'redux-persist/lib/storage' // defaults to localStorage for web

// persist config of persistStore, which enables me to store my redux even if I reload the page
const persistConfig = {
    key: 'root',
    storage,
}

const persistedReducer = persistReducer(persistConfig, rootReducer);

const store = createStore(
    persistedReducer,
    // extension to be more easy to check redux, to see it install redux devtools in chrome browser
    window.__REDUX_DEVTOOLS_EXTENSION__ && window.__REDUX_DEVTOOLS_EXTENSION__()
);
const persistor = persistStore(store);
export { store, persistor };