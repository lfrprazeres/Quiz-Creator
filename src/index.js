import { Provider } from "react-redux";
import React from 'react';
import ReactDOM from 'react-dom';
import App from './App';
import { store, persistor } from './store';
import './global.css';
import { PersistGate } from 'redux-persist/integration/react';

ReactDOM.render(
    <Provider store={store}>
      <PersistGate loading={null} persistor={persistor}>
        <App />
      </PersistGate>
    </Provider>,
  document.getElementById('root')
);