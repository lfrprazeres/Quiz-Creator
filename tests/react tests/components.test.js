import React from 'react';
import { shallow, configure } from 'enzyme';
import { 
    CreateSection,
    Button,
    QuizTable,
} from '../../src/components';
import Adapter from 'enzyme-adapter-react-16';
import { Provider } from "react-redux";
import { store } from '../../src/store';
import toJson from 'enzyme-to-json'

configure({adapter: new Adapter()});

describe('Components tests', () => {
    it('should render CreateSection without crash with the properly props', () => {
        const props = {
            title: "component title",
            current:"1",
            total:"2",
            buttonText: "Next",
            backTitle: "Back to List",
            onClickButton: jest.fn()
        };
        const wrapper = shallow(
            <Provider store={store}>
                <CreateSection {...props} />
            </Provider>
            );
        expect(toJson(wrapper)).toMatchSnapshot();
    });

    it('should render Button without crash with the properly props', () => {
        const props = {
            variant:"contained", 
            color:"primary", 
            className: "className",
            onClick:jest.fn(),
            bg:"blue"
        };
        const wrapper = shallow(<Button {...props} />);
        expect(toJson(wrapper)).toMatchSnapshot();
    });

    it('should render QuizTable without crash with the properly props', () => {
        const props = {
            quiz: [
                [ 1, "title 1", "description 1" ],
                [ 2, "title 2", "description 2" ],
                [ 3, "title 2", "description 3" ]
            ],
            onDeleteQuiz: jest.fn()
        }
        const wrapper = shallow(
            <Provider store={store}>
                <QuizTable {...props} />
            </Provider>
        );
        expect(toJson(wrapper)).toMatchSnapshot();
    });
})