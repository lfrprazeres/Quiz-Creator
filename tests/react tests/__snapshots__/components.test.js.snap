// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components tests should render Button without crash with the properly props 1`] = `
<Styled(WithStyles(ForwardRef(Button)))
  bg="blue"
  className="className"
  color="primary"
  onClick={[MockFunction]}
  variant="contained"
/>
`;

exports[`Components tests should render CreateSection without crash with the properly props 1`] = `
<ContextProvider
  value={
    Object {
      "store": Object {
        "dispatch": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
        Symbol(observable): [Function],
      },
      "subscription": Subscription {
        "handleChangeWrapper": [Function],
        "listeners": Object {
          "notify": [Function],
        },
        "onStateChange": [Function],
        "parentSub": undefined,
        "store": Object {
          "dispatch": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
          Symbol(observable): [Function],
        },
        "unsubscribe": null,
      },
    }
  }
>
  <Connect(CreateSection)
    backTitle="Back to List"
    buttonText="Next"
    current="1"
    onClickButton={[MockFunction]}
    title="component title"
    total="2"
  />
</ContextProvider>
`;

exports[`Components tests should render QuizTable without crash with the properly props 1`] = `
<ContextProvider
  value={
    Object {
      "store": Object {
        "dispatch": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
        Symbol(observable): [Function],
      },
      "subscription": Subscription {
        "handleChangeWrapper": [Function],
        "listeners": Object {
          "notify": [Function],
        },
        "onStateChange": [Function],
        "parentSub": undefined,
        "store": Object {
          "dispatch": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
          Symbol(observable): [Function],
        },
        "unsubscribe": null,
      },
    }
  }
>
  <Connect(QuizTable)
    onDeleteQuiz={[MockFunction]}
    quiz={
      Array [
        Array [
          1,
          "title 1",
          "description 1",
        ],
        Array [
          2,
          "title 2",
          "description 2",
        ],
        Array [
          3,
          "title 2",
          "description 3",
        ],
      ]
    }
  />
</ContextProvider>
`;
